peer/brontide.go

```diff
@@ -2636,6 +2629,55 @@ func (p *Brontide) handleLinkFailure(failure linkFailureReport) {
 	}
 }
 
+// tryLinkShutdown attempts to fetch a target link from the switch, calls
+// ShutdownIfChannelClean to optimistically trigger a link shutdown, and
+// removes the link from the switch. It returns an error if any step failed.
+func (p *Brontide) tryLinkShutdown(cid lnwire.ChannelID) error {
```

Comment:
Not familiar with the codebase but is the newly added logic already tested by the old tests?

Reply:
no it's not, could be done with mock links tho

Reply:
It is currently used in tests though, I misread your comment.  Could add mock assertions

Reply:
I wasn't sure how to do this with `mock.Mock` and `mockMessageSwitch`.  What would we assert?  There are several calls to the mock switch in unit tests, I'm just not entirely sure what we'd assert - that the functions themselves were called? That a "link" is only returned if it exists in the slice of links via GetLinksByInterface?  That's my current blocker here